.github/workflows/manual_delivery.yaml

```diff
@@ -0,0 +1,105 @@
+name: Manual Docker Build, Tag & Push
+
+on:
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: 'Application Environment'     
+        required: true
+        default: 'dev'
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    outputs:
+      ENV: ${{ steps.tag.outputs.env }}
+      TAG: ${{ steps.tag.outputs.tag }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set Tag
+        id: tag
+        run: |
+          export CI_COMMIT_SHORT_SHA=$(git describe --abbrev=7 --always --tags)
+          echo "::set-output name=tag::$CI_COMMIT_SHORT_SHA"
+          echo "Commit for this pipeline: ";
+          echo $CI_COMMIT_SHORT_SHA;
+          export ENV=${{ github.event.inputs.environment }}
+          echo "::set-output name=env::$ENV"
+          echo "Environment for this pipeline: ";
+          echo $ENV
+  build-push-image:
+    runs-on: ubuntu-latest
+    needs: prepare
+
+    services:
+      postgres:
+        image: postgres:latest
+        env:
+          POSTGRES_PASSWORD: discourse
+          POSTGRES_PORT: 5432
+          POSTGRES_USER: discourse
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      redis:
+        image: redis
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Database per CI Environment
+        id: env_databases_setup
+        env:
+          ENV: ${{ needs.prepare.outputs.ENV }}
+        run: |
+          set -e
+          set -u
+          echo "Creating user and database 'discourse-$ENV'"
+          psql -v ON_ERROR_STOP=1 "postgresql://discourse:discourse@localhost:5432" <<-EOSQL
+              CREATE DATABASE "discourse-$ENV";
+              GRANT ALL PRIVILEGES ON DATABASE "discourse-$ENV" TO discourse;
+          EOSQL
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        id: configure-aws-creds
+        with:
+          aws-access-key-id: ${{ secrets.ECR_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.ECR_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+      - name: Login to Amazon ECR
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+      - name: Build and Push Stage Image to ECR
+        id: build-push
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: discourse
+          ENV: ${{ needs.prepare.outputs.ENV }}
+          TAG: ${{ needs.prepare.outputs.TAG }}
+        run: |
+          export GHA_NETWORK=$(docker network ls --filter driver=bridge --filter name=github_network_ --format '{{.Name}}')
```

Comment:
We have some repitition here, but after learning last time how painful it can be to factor this out let's just keep it this way this time. :)